### PR TITLE
Fixes memory leakes in Filewatch test when using ASAN

### PR DESCRIPTION
**Pull request by tayloryoung6396**

### CODE REVIEW

**Review by Bidski — APPROVED** [review]
Since I wrote this, I guess its fine

File: module/extension/FileWatcher/src/FileWatcher.cpp [hunk 1]
```diff
@@ -290,7 +291,24 @@ namespace module::extension {
...
+            if (remove) {
+                uv_fs_event_stop(remove.get());
+                uv_close(reinterpret_cast<uv_handle_t*>(remove.get()), [](uv_handle_t*) {});
+            }
+        }
+        uv_close(reinterpret_cast<uv_handle_t*>(remove_watch.get()), [](uv_handle_t*) {});
+        uv_close(reinterpret_cast<uv_handle_t*>(add_watch.get()), [](uv_handle_t*) {});
+        uv_close(reinterpret_cast<uv_handle_t*>(shutdown.get()), [](uv_handle_t*) {});
+        while (uv_run(loop.get(), UV_RUN_NOWAIT) != 0) {
+        }
```
Comment (TrentHouliston) [inline module/extension/FileWatcher/src/FileWatcher.cpp:311]: Does it work if you put this up the top at line 294? And if you do does it deal with the remove_queue?

Comment (Bidski) [reply module/extension/FileWatcher/src/FileWatcher.cpp:311]: nope

Comment (Bidski) [reply module/extension/FileWatcher/src/FileWatcher.cpp:311]: seems to hang

Comment (tayloryoung6396) [reply module/extension/FileWatcher/src/FileWatcher.cpp:311]: Since someone made me remove 19 other files... i have 278 things to compile

Comment (Bidski) [reply module/extension/FileWatcher/src/FileWatcher.cpp:311]: anywhere other than after closing the asyncs causes the test to hang (didn't test after the loop close)

**Pull request merged**